pkg/queue/request_metric_test.go

```diff
@@ -123,6 +124,42 @@ func TestRequestMetricHandlerPanickingHandler(t *testing.T) {
 	handler.ServeHTTP(resp, req)
 }
 
+func BenchmarkNewRequestMetricHandler(b *testing.B) {
+	baseHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+	})
+	breaker := NewBreaker(BreakerParams{QueueDepth: 10, MaxConcurrency: 10, InitialCapacity: 10})
+	stat := &fakeStatsReporter{}
```

Comment:
Can we use the actual stats reporter here? It's been a source of non-trivial allocations and we're improving on that currently. Having the "real deal" in here will help surface that.